### PR TITLE
Enclosed volume harpoon object upgrade

### PR DESCRIPTION
**Pull request by BingWangUS**
### Issues addressed by this PR

Closes #

### Changelog

 - add a custom upgrader for HarpoonObject in CFD_Toolkit

### CODE REVIEW

**Comment by BingWangUS** [comment]
@BHoMBot check all

File: BHoMUpgrader41/Converter.cs [hunk 1]
```diff
@@ -155,6 +156,32 @@ public Converter() : base()
...
+                    newVersion["_t"] = newVersion["_t"].ToString().Replace("HarpoonObject", "SurfaceHarpoonObject");
+                    newVersion.Remove("Center");
+                }
+                else
+                {
+                    newVersion["_t"] = newVersion["_t"].ToString().Replace("HarpoonObject", "VolumeHarpoonObject");
+                    newVersion.Add("Location", newVersion["Center"]);
+                    newVersion.Remove("Center");
+                }
+            }
```
Comment (FraserGreenroyd) [inline BHoMUpgrader41/Converter.cs:179]:
```suggestion
            if (newVersion.ContainsKey("Center") && newVersion["Centre"] != null)
            {
                    newVersion["_t"] = newVersion["_t"].ToString().Replace("HarpoonObject", "VolumeHarpoonObject");
                    newVersion.Add("Location", newVersion["Center"]);
            }
            else
                    newVersion["_t"] = newVersion["_t"].ToString().Replace("HarpoonObject", "SurfaceHarpoonObject");

            newVersion.Remove("Center");
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Probably need to rebase to get the changes I put in with @michaelhoehn recently (apologies!). Suggested change is a bit cleaner and makes sure the `newVersion` gets a type upgrade regardless of whether it contains a key for `center` or not :smile:

We also need to add versioning for the deleted method to the JSON with the message we discussed :smile:

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Review by FraserGreenroyd — APPROVED** [review]
This works with the other PR and passes versioning.

**Pull request merged**